libs/remote_storage/src/lib.rs

```diff
@@ -43,6 +42,9 @@ pub use self::{
 };
 use s3_bucket::RequestKind;
 
+/// Azure SDK's ETag type is a simple String wrapper: we use this internally instead of repeating it here.
+pub use azure_core::Etag;
```

Comment:
nit: This is fine because this type just wraps a string, but etags can differ across cloud providers, right? What if azure changes their etag to something funky in the future and the s3 etag isn't compatible with it? This is more of a philosophical concern though :)

Reply:
Philosophical indeed :-)  Ultimately these are HTTP headers, i.e. strings.  On S3 they're _almost_ md5 digests, unless it's a multipart upload.  For our purposes they're juts something we need to be able to do an equality on.